host/http.go

```diff
@@ -42,16 +41,8 @@ func (h *Host) streamEvents(id string, w http.ResponseWriter) error {
 		<-w.(http.CloseNotifier).CloseNotify()
```

Comment:
I'd also abstract the `CloseNotify` away and specify a cleanup function on the stream, like in my proposal [here](https://github.com/flynn/flynn/issues/744#issuecomment-70214292). I'd also probably simplify the implementation to just be a single blocking function -- since the channel we pass in is of type `chan interface{}`, to log out errors, we could simply detect the error type using reflection, and then write them out as an error.